### PR TITLE
Gross net spending polish

### LINKED ISSUE #678: Gross and Net Planned Spending table needs polish
image

 - Inconsistency between "Planned Spending (net)" and "Gross planned spending", both in structure and capitalization
    - this may have been partially on purpose, because net is what we usually mean when we say planned spending and this may have been done to try and communicate it that way (but it just looks sloppy)
  - Ordering feels off. With this ordering, column 1 = columns 2 + 3 - 4, but I think you'd expect the final column to be the product, not the first. Maybe order it either `gross | special | revenue | net` or `revenue | gross | special | net`?
    - could be good to some how further communicate the formula happening in this table visually (make revenue numbers red, net or gross some third colour?)
  - "Planned Specified Purpose Account Spending" should be hidden if it's zero in all years, it's a weird special case that very few departments have if I remember right (someone look at the data and check me on that). Annoying to have one super wide column in the middle of the table if it's going to be a bunch of zeros more often than not

### PR DESCRIPTION
**Pull request by GarvinH**
closes #678 

CBSA
image

JUS
image

Fixed capitalization
Rearranged columns to be cleaner
Hide SPA if it is all 0

### CODE REVIEW

**Comment by Stephen-ONeil** [comment]
Just mentioning this in passing, haven't checked my self, but I'm not sure that shade of green is colour-blind accessible. Check the style applying that colour in the chrome dev tools, it will tell you the contrast ratio (and point out if it isn't a high enough contrast).

File: client/src/components/DisplayTable/DisplayTable.js [hunk 1]
```diff
@@ -92,6 +92,7 @@ export class DisplayTable extends React.Component {
           sum_func: (sum, value) => ... <- (function) Custom sum func. Default to sum + value
           sort_func: (a, b) => ... <- (function) Custom sort func. Default to _.sortBy
           sum_initial_value: 0 <- (number) Default to 0
+          color: "class name"
```
Comment (ktw1016) [inline client/src/components/DisplayTable/DisplayTable.js:95]: This config can be a lot more flexible. With how it's written it is:
1. only limited to style color of td
2. only styles specific color
3. What if in the future, I want to use textGreen but it has nothing to do with `spending`?

You probably get my point. so this can be written with a lot more flexibility.

If you want to have additional config dedicated to `td` style, we can have, `td_class_name` (maybe think of a better name if that sounds bad). It would be a string, which is a css classname defined in where DisplayTable is called. That way, whoever is calling it will need to specify the color. Furthermore, it can be any css rule, not limited to only the color.

Comment (ktw1016) [reply client/src/components/DisplayTable/DisplayTable.js:95]: BUT, I'd personally prefer to actually even not do that haha. For a reason where it's kind of a duplicate of `formatter` config, and I want to avoid cluttering up the configs and keep it cleaner as much as possible.

so with formatter, you can do something like:
`formatter: (value) => <Format style={{color: 'red'}} type={"dollar"} content={value} />`
okay but then `Format` probably won't take style attribute so maybe you can add that, or do some workaround!

File: client/src/panels/panel_declarations/finances/drr_dp_resources/dp_rev_split.js [hunk 2]
```diff
@@ -71,27 +72,39 @@ export const declare_dp_rev_split_panel = () =>
...
+            color: "spending",
+            initial_visible: (() => {
+              for (const data of table_data) {
+                if (data.spa != 0) {
+                  return true;
+                }
+              }
+
+              return false;
+            })(),
```
[the remark this answers is not in the document]
Comment (Stephen-ONeil) [reply client/src/panels/panel_declarations/finances/drr_dp_resources/dp_rev_split.js:96]: Or `_.every` https://lodash.com/docs/4.17.15#every

File: client/src/components/DisplayTable/DisplayTable.js [hunk 3]
```diff
@@ -117,7 +117,7 @@ export class DisplayTable extends React.Component {
       const current_col_formatter = col_configs_with_defaults[col].formatter;
       const current_col_raw_formatter =
         col_configs_with_defaults[col].raw_formatter;
-      if (current_col_formatter) {
+      if (current_col_formatter && !_.isFunction(current_col_formatter)) {
```
Comment (ktw1016) [inline client/src/components/DisplayTable/DisplayTable.js:120]: Just wondering why did you need this one?

File: client/src/panels/panel_declarations/finances/drr_dp_resources/dp_rev_split.js [hunk 4]
```diff
@@ -8,6 +8,8 @@ import {
   get_source_links,
   declare_panel,
 } from "../../shared.js";
+import { Format } from "../../../../components";
+import styles from "../../../../common_css/_common-variables.scss";
```
Comment (ktw1016) [inline client/src/panels/panel_declarations/finances/drr_dp_resources/dp_rev_split.js:12]: Doesn't seem used

File: client/src/panels/panel_declarations/finances/drr_dp_resources/dp_rev_split.js [hunk 5]
```diff
@@ -22,6 +24,14 @@ const special_cols = _.flatMap(planning_years, (year) => [
 ]);
 const dp_cols = [...planning_years, ...special_cols];
 
+const spending = (value) => (
+  <Format style={{ color: "#008000" }} type={"dollar"} content={value} />
```
Comment (ktw1016) [inline client/src/panels/panel_declarations/finances/drr_dp_resources/dp_rev_split.js:28]: you can throw these in `infobase_color_constants`

File: client/src/components/DisplayTable/DisplayTable.js (hunk 1 above)
Comment (ktw1016) [inline client/src/components/DisplayTable/DisplayTable.js:95]: You didn't delete this one

File: client/src/components/DisplayTable/DisplayTable.scss [hunk 6]
```diff
@@ -89,6 +93,14 @@ table.display-table:not(.no-total-row) > tbody > tr:last-child > td:not(:last-ch
...
 }
 
+td.spending {
+  color: $textGreen;
+}
+
+td.revenue {
+  color: $textRed;
+}
+
```
Comment (ktw1016) [inline client/src/components/DisplayTable/DisplayTable.scss:103]: You didn't delete these

File: client/src/panels/panel_declarations/finances/drr_dp_resources/dp_rev_split.js (hunk 5 above)
Comment (ktw1016) [reply client/src/panels/panel_declarations/finances/drr_dp_resources/dp_rev_split.js:28]: Actually now that I think of it, green and red colors actually are already defined. Check `color_defs.js`, you can get these from `window. infobase_color_constants`

File: client/src/components/DisplayTable/DisplayTable.js (hunk 3 above)
Comment (GarvinH) [reply client/src/components/DisplayTable/DisplayTable.js:120]: Since I changed formatter to a function, it would float left since `_.isString(current_col_formatter)` returned false.

File: client/src/panels/panel_declarations/finances/drr_dp_resources/dp_rev_split.js (hunk 5 above)
Comment (GarvinH) [reply client/src/panels/panel_declarations/finances/drr_dp_resources/dp_rev_split.js:28]: The green in color_defs.js were too low contrast, so I took the green and red from the estimates comparison

File: client/src/components/DisplayTable/DisplayTable.js (hunk 3 above)
Comment (ktw1016) [reply client/src/components/DisplayTable/DisplayTable.js:119]: This seems poorly written originally. Try this:
```     
return (
        (_.isString(current_col_formatter) && _.isNumber(row[col])) ||
        _.isNumber(current_col_raw_formatter(row[col]))
  ? "right" : "left"
)
```

File: client/src/core/color_defs.js [hunk 7]
```diff
@@ -20,6 +20,8 @@ const buttonSecondaryColor = "#76a4df";
 
 const textColor = "#333";
 const textLightColor = "#fff";
+const textGreen = "#008000";
+const textRed = "#cc0000";
```
[the remark this answers is not in the document]
Comment (GarvinH) [reply client/src/core/color_defs.js:24]: I used the highlight red and the success green. the red works but success green doesn't. The issue has to do with contrast against the white background. Stephen told me to use these colors. I've made sure to add that to the scss though.

File: client/src/components/DisplayTable/DisplayTable.scss [hunk 8]
```diff
@@ -1,5 +1,9 @@
 @import '../../common_css/common-variables';
 
+td {
```
[the remark this answers is not in the document]
Comment (GarvinH) [reply client/src/components/DisplayTable/DisplayTable.scss:3]: That was there because chrome was detecting the the blue background instead of the white background causing incorrect contrast ratio being displayed.

File: client/src/panels/panel_declarations/finances/drr_dp_resources/dp_rev_split.js [hunk 9]
```diff
@@ -23,17 +23,17 @@ const special_cols = _.flatMap(planning_years, (year) => [
 ]);
 const dp_cols = [...planning_years, ...special_cols];
 
-const spending = (value) => (
+const spending = (value, custom_color = null) => (
```
Comment (ktw1016) [inline client/src/panels/panel_declarations/finances/drr_dp_resources/dp_rev_split.js:26]: This can actually be better named like `spending_formatter`

File: client/src/components/DisplayTable/DisplayTable.js [hunk 10]
```diff
@@ -400,7 +400,7 @@ export class DisplayTable extends React.Component {
                               } else {
                                 return col_configs_with_defaults[
                                   col_key
-                                ].formatter(total_row[col_key]);
+                                ].formatter(total_row[col_key], "white");
```
Comment (ktw1016) [inline client/src/components/DisplayTable/DisplayTable.js:403]: I think we can safely make the text white across all display tables

so try putting `!important` on line 89 in Displaytable.scss. You'll probably need to work around with inline css on your formatters
